feat(CLI): Automatically expand loaded env variables

d-asensio opened the pull request:
<!-- ⚠️⚠️ Acknowledge ALL below remarks -->
<!-- ⚠️⚠️ PR will not be processed if it doesn't meet outlined criteria -->

<!-- ⚠️⚠️ Do not propose PR's without prior agreement on a solution in the corresponding issue -->
<!-- ⚠️⚠️ Only documentation updates and obvious bug fixes are welcome without it -->

<!--
⚠️⚠️ Ensure to follow code style guidelines
https://github.com/serverless/serverless/blob/master/CONTRIBUTING.md#code-style
-->

<!--
⚠️⚠️ Ensure to cover changes with tests written according to test guidelines
https://github.com/serverless/serverless/blob/master/test/README.md
-->

<!-- ⚠️⚠️ Ensure that support for Node.js v10 is maintained. -->

<!--
⚠️⚠️ Ensure that the proposed change passes CI. Confirm that by running the following scripts:
• npm run prettier-check
• npm run lint
• npm test
-->

<!--
⚠️⚠️ If proposed change touches integration with AWS services, confirm integration tests pass:
https://github.com/serverless/serverless/blob/master/test/README.md#aws-integration-tests
-->

<!-- ⚠️⚠️ After your PR is submitted, review the final CI status and address eventual failure -->

<!-- ⚠️⚠️ Answer below questions -->

<!--
Q1: Provide a link to the corresponding issue

• If PR *partially* addresses issue, ensure to rename "Closes" to "Addresses" ("Closes" term will automatically close an issue on PR merge)
• If it's a documentation update or obvious bug fix that has no corresponding issue, replace this line with a short description of made changes
-->

Closes: #9689

pgrzesik commented:
Hello @d-asensio - could you please include motivation behind this PR proposal? I believe there is not corresponding issue created for that , correct?

d-asensio commented:
Hey @pgrzesik sure! I found some time to write an issue with the motivations of this. I referenced it in my first comment.

Thank you!

pgrzesik reviewed (commented):
Thanks a lot @d-asensio - it looks great overall and thanks so much for writing up a motivation in a linked issue. I think it's a good idea to allow such expansions, but it would also be good to document it e.g. here: https://github.com/serverless/serverless/blob/master/docs/environment-variables.md

Btw - it looks like there's a conflicting file - could you please look into that and rebase your branch?

d-asensio commented:
Hey, @pgrzesik I rebased it onto master and added a comment with a little example in the docs.

Now the dependencies security check is failing, but I think is due to the rebase, since [in master it is failing as well](https://snyk.io/test/github/serverless/serverless#SNYK-JS-ADMZIP-1065796). In any case, I don't have permission to check snyk for this branch, could you help me with this?

Thank you!

d-asensio commented:
Hi @pgrzesik, is there something missing here?

pgrzesik commented on docs/environment-variables.md:
```diff
@@ -12,6 +12,17 @@ With `useDotenv: true` set in your `serverless.yml` file, framework automaticall
 
 The framework looks for `.env` and `.env.{stage}` files in service directory and then tries to load them using `dotenv`. If `.env.{stage}` is found, `.env` will not be loaded. If stage is not explicitly defined, it defaults to `dev`.
 
+### Variable expansion
```
Let's also mention what specific library is being used to perform this expansion

pgrzesik reviewed (commented):
Thanks a lot :bow: Sorry for the delay in review - I missed the notification and then I was off for a few days. I have one minor suggestion related to docs - also, let's wait for opinion from @medikoo if we should introduce such extensions.

medikoo reviewed (approved):
@d-asensio @pgrzesik it feels as great addition to me.  It mimicks how things will work in a shell, which is a nice improvement.

@pgrzesik After adjustment to docs is made, feel free to merge

pgrzesik commented:
Thanks @medikoo :bow: @d-asensio so the only thing to address is one comment about attributing what library is used for expansions and we're good to go :+1:

d-asensio commented:
Done! Thank you both @pgrzesik and @medikoo! 🥳 🎉

pgrzesik reviewed (approved):
Thank you @d-asensio, well done :clap:
